backend/src/swa-tracker/swa-tracker-wrapper.ts

```diff
@@ -20,89 +20,97 @@ const EVENT_TYPES = {
- *
- * @returns { SWATracker }
- */
-export function getSWA(): SWATracker {
-	if (isInitialized() === false) {
-		throw Error(ERROR_ANALYTICS_TRACKER_NOT_INITIALIZED);
+	private static isInitialized(): boolean {
+		return (this.swaTracker !== undefined ) ? true : false;
+	}
+	
+	private static initSWATracker(newSwaTracker: SWATracker) {
+		this.swaTracker = newSwaTracker;
+	}
+	
+	
+	/**
+	 * Note the use of a getter function so the value would be lazy resolved on each use.
+	 * This enables concise and simple consumption of the swaTracker throughout our Extension.
+	 *
+	 * @returns { SWATracker }
+	 */
+	private static getSWA(): SWATracker {
+		if (this.isInitialized() === false) {
+			throw Error(ERROR_ANALYTICS_TRACKER_NOT_INITIALIZED);
+		}
+		return this.swaTracker;
 	}
-	return swaTracker;
-}
 
-export function createSWATracker() {
-	const swaTracker = new SWATracker(
-		"SAPSE",
-		YEOMAN_UI,
-		// callback for error, one such callback for all the errors we receive via all the track methods
-		// error can be string (err.message) or number (response.statusCode)
-		(error: string | number) => {
-			if (typeof error === 'string') {
-				getLogger().error("SAP Web Analytics tracker failed to track", {errorMessage: error});
-			} else if (typeof error === 'number') {
-				// bug in matomo-tracker: they think that success is 200 or 30[12478], so we are rechecking here
-				if (!((error >= 200) && (error <= 299)) && !((error >= 300) && (error <= 399))) {
-					getLogger().error("SAP Web Analytics tracker failed to track", {statusCode: error});
+	public static createSWATracker() {
+		const swaTracker = new SWATracker(
+			"SAPSE",
+			YEOMAN_UI,
+			// callback for error, one such callback for all the errors we receive via all the track methods
+			// error can be string (err.message) or number (response.statusCode)
+			(error: string | number) => {
+				if (typeof error === 'string') {
+					getLogger().error("SAP Web Analytics tracker failed to track", {errorMessage: error});
+				} else if (typeof error === 'number') {
+					// bug in matomo-tracker: they think that success is 200 or 30[12478], so we are rechecking here
+					if (!((error >= 200) && (error <= 299)) && !((error >= 300) && (error <= 399))) {
+						getLogger().error("SAP Web Analytics tracker failed to track", {statusCode: error});
+					}
 				}
 			}
+		  );
+	
+		// Update the swa-tracker-wrapper with a reference to the swaTracker.
+		this.initSWATracker(swaTracker);
+		getLogger().info(`SAP Web Analytics tracker was created for ${YEOMAN_UI}`);
+	}
+	
+	
+	public static updateGeneratorStarted(generatorName: string) {
+		const eventType = EVENT_TYPES.PROJECT_GENERATION_STARTED;
+		let customEvents = [generatorName];
+		let startTime = this.startTime = Date.now();
+		this.getSWA().track(eventType, customEvents);
+		getLogger().trace("SAP Web Analytics tracker was called and start time was initialized", {
+			eventType, generatorName, startTime, customEvents});
```

Comment:
We don't need local startTime variable.
Inside the json sent to logger: startTime: this.startTime